Iterations on options modal

karmatosed opened the pull request:
A few iterations to the options modal in line with #24965 as a step for this release, whilst the new design is being worked on.

This PR is starting to focus on the points highlighted in the comment by @mtias:

> Improve the copy of each setting, make them more concrete and cohesive (some are many words long, some just a few)
Utilize longer descriptions when necessary (similar to "Top Toolbar") to provide more context.
Reorganize the main sections.

## Changes
I have done some copy changes and worked on bringing in the same style for help text, however, for now, kept the checkboxes. One point to note that could be done if copy is agreed in time is moving also to toggles for this release.

Here you can see a visual of what this PR results in:

![image](https://user-images.githubusercontent.com/253067/95097359-4c497900-0725-11eb-9762-7c59abaff9ae.png)

## Feedback
I don't consider this to be a final PR, I would love feedback and here are a few areas to focus that, although general feedback is as always welcome.

- Clarity: do the changes bring clearer definitions? What changes are missing and should be brought in?
- Uniformity of text: does this help with the variations?

I would also love a code review as there is some style adding and I need to be sure doing that correctly in this case. Work will continue with the design over on the issue, but let's get something in for this release as an iterative step. Thanks in advance.

## Next steps
Once I have feedback, I want to work to within a few days getting something merged for this as we can always iterate from this point. 

## Thanks
Props to @jasmussen and @ItsJonQ for helping me working through some code hunting and setup fun to get this PR brewed.

Commit:
Iterations on options modal

A few iterations to the options modal in line with #24965 as a step for this release, whilst the new design is being worked on.

ItsJonQ commented:
@karmatosed Thanks for working on this!

The code implementation looks good to me 👍 

Adding the `.scss` code for direct targeting of the helpText is probably the best solution we have at the moment.

Zoom into the UI a bit... I feel like maybe the margins could be tweaked just a tiny bit?

I have a screenshot of the helpText margins tweaked:

<img width="444" alt="Screen Shot 2020-10-05 at 11 48 29 AM" src="https://user-images.githubusercontent.com/2322354/95101932-bf8cc400-0700-11eb-95e4-27e700fbd07e.png">

The final style was `margin: -8px 0 8px 42px` (it was something I did quickly in Chrome).

What do you think :)

Commit:
Iterates margins

Commit:
Increase margin a little more to adjust

karmatosed commented:
Great idea, I iterated a little and pushed that commit. Thanks for taking a look @ItsJonQ. Here is where I came to:

![image](https://user-images.githubusercontent.com/253067/95113041-b5d38280-0739-11eb-80a9-b50db5f60b5d.png)

ItsJonQ commented:
@karmatosed Great :)

Peeking at the CSS variables, my initial suggestion was for something like this:

```
margin: -$grid-unit-10 0 $grid-unit-10 ( $grid-unit-50 + 2px );
```

That aside... The tests seem a bit unhappy 🙈 .
I'm not sure about the E2E one. But I can help with the JavaScript unit ones.

Lemme know if you some help with those 🙏

jasmussen commented:
Nicely done. Looking at this one:

<img width="432" alt="Screenshot 2020-10-06 at 08 34 37" src="https://user-images.githubusercontent.com/1204802/95166777-c96d0080-07ae-11eb-8b8b-fa308592b680.png">

The spacing below the help text looks more correct in the previous screenshot shared, not sure which one is the latest:

<img width="448" alt="Screenshot 2020-10-06 at 08 34 14" src="https://user-images.githubusercontent.com/1204802/95166803-d25dd200-07ae-11eb-95f1-abd6717bfc7f.png">

Also should the helptext be not italic, like in the dropdown?

<img width="307" alt="Screenshot 2020-10-06 at 08 34 10" src="https://user-images.githubusercontent.com/1204802/95166823-da1d7680-07ae-11eb-8190-1de9b15d9752.png">

Nice bit of constency to be found in unifying those two. Cool!

Commit:
Iterate margin calculation and also remove italics

Props @jasmussen and @ItsJonQ

karmatosed commented:
Thanks @ItsJonQ and @jasmussen - you are both really helping this take shape! 

@jasmussen you rightly pointed out after this PR there's a win to be had here in another one for a fix on help text to unify, there's an italic being picked up that for now I've set an override on, but can fix upstream in another PR (I'll get that begun outside this). 

Here is the current state of PR:

![image](https://user-images.githubusercontent.com/253067/95173978-57e28180-07b1-11eb-865c-dc049b22c649.png)

@ItsJonQ that calculation works nicely, thank you for that refinement. I would love help sorting the tests, I'm not totally convinced all are valid either which is perplexing, so would love to work through those. Thank you for offering to help there.

karmatosed commented:
Linking in #25851 as that has the root fix for the font styling so once in can remove the work around in this PR.

Commit:
Update snapshot test for Options modal

ItsJonQ commented:
Taking a look at the E2E test failures... I'm really not sure why they're breaking 🤔

ItsJonQ commented:
@karmatosed I rebased it with latest `master` to see if that will help with tests 🙏

Commit:
Shorten text

Shortens the length of text.

enriquesanchez commented:
Took this for a spin:

<img width="450" alt="Screen Shot 2020-10-06 at 14 02 19" src="https://user-images.githubusercontent.com/3276087/95248226-9c374700-07dc-11eb-891c-9509b4c1940a.png">

I really like this direction and the changes you've made to the margins and style of the helper text 👍

I have some suggestions for some of the helper texts. Please feel free to ignore if these don't make sense to you:

For "Pre-publish checklist", how about "Allows review of document settings before publishing."?

For "Contain the text cursor", how about "Keeps keyboard cursor inside active block."?

For 'Compact interface', how about "Reduces options in toolbar and outlines."?

ntsekouras committed:
fix e2e tests

ntsekouras committed:
fix snapshot

ntsekouras reviewed (commented):
Hey @karmatosed - thanks! I pushed a couple of commits to fix the failing tests.

karmatosed commented:
Thanks so much @ntsekouras for your help there.

@enriquesanchez those look like some great suggestions, let me work on an iteration. Thank you.

Commit:
Iterates help text

Commit:
Update index.js

karmatosed commented:
One other potential iteration I have been working on is to change up the sections. @enriquesanchez I would love your input here if that is possible as I know you are working on a keyboard shortcut section, that to me feels like this option doesn't fit there (although it might). In the explorations for the iterated options panel, the sections included 'blocks' and this could be something explored here:

![image](https://user-images.githubusercontent.com/253067/95369643-9cfcd580-08cf-11eb-88a4-03772a2bf052.png)

jasmussen commented:
Not bad. Not for this PR, but what would it look like without those borders, perhaps slightly more space instead?

Commit:
Try removal of borders

Commit:
Increase bottom margin

enriquesanchez commented:
I agree that a whole section just for the option of 'Contain the text cursor' feels like too much, but I also don't find that it fits with any of the others. 

Have we considered having an 'Accessibility' section in here? I think both 'Contain the text cursor' and 'Display button text labels' could fit well under it.

karmatosed commented:
@jasmussen I think you have a great point there and maybe a middle ground is to remove the borders for now in the content area, leave the header? I have pushed a version of that to the PR just to see, happy for feedback around that as it's something we can always reverse out if want to.

Here is what I ended up with adding a small amount of spacing increase:

![image](https://user-images.githubusercontent.com/253067/95383015-67151c80-08e2-11eb-9259-d386c471c78c.png)

jasmussen commented:
I like the reduced borders there! Seems good to me 👍 👍

Commit:
Set to use help text variable

ItsJonQ reviewed (approved):
🚀 from me! Thanks for helping with this @karmatosed + @jasmussen

ZebulanStanphill commented:
Now that #25683 is merged, this needs a rebase.

karmatosed commented:
Noting that above is rebasing so it's in sync. Thanks for reminder @ZebulanStanphill.

carolinan commented:
Just looking at the images presented here, I don't understand what "_Most used blocks in library"_ is 
...Do what with the most used blocks?

ZebulanStanphill commented:
@carolinan Yeah, the wording seems a bit unclear to me as well. I think "Show Most Used section in block library" or something like that would be clearer.

karmatosed commented:
@carolinan thanks for the suggestion. The idea was the toggle would be the activator there. I am very happy to add back in 'show' but do you feel the same for the button text label also? I think both of those being consistent makes sense.

kellychoffman commented:
Much cleaner look without the border. 

The copy is still rather hard to parse tho. I would take another pass to make the labels more descriptive, and where its needed, add helper text. I also think you could consider adding helper text to some or all of the section headers. (For example, I don't think everyone knows what or where the "Document panel" is.) 

I'd be careful that the helper text is not just rephrasing the label. "Contain the text cursor" is one example of this where the helper text is not necessarily clarifying. Maybe the helper text could instead say _why_ you'd show/hide this.

Commit:
Iterate copy and add in new description to sections.

Commit:
Iterate description text

Commit:
Update index.js

karmatosed commented:
I took the feedback and iterated, here you can see the results:

![image](https://user-images.githubusercontent.com/253067/95901806-cb752780-0d8b-11eb-8195-c8204f43a344.png)


Things I changed:

- A new class and descriptive text to section component.
- Iterated almost all text.
- Changed some headings.

Huge thanks to @enriquesanchez for working with me on the copy and to @ItsJonQ for guidance on adding to the component descriptive text, running tests.

@kellychoffman I would love to get your opinion on this as feel it's in a place to at least get in for release. Would love your thoughts there.

@ItsJonQ would love a code review once the tests have finished if that is possible.

ZebulanStanphill commented:
I think "preventing text caret leaving blocks" reads weirdly... I think "preventing text caret **from** leaving blocks" would be better.

We should probably note that "Use theme styles" may have no effect if the theme doesn't support the feature.

The term "post panel" isn't used anywhere else, and may not make as much sense on other post types.

Other than those three things, I like this. It's certainly an improvement over `master`.

ItsJonQ commented:
👍 for the new `description` prop implementation for the `Section` component

karmatosed commented:
@ZebulanStanphill Thanks for your input. Regarding:

> We should probably note that "Use theme styles" may have no effect if the theme doesn't support the feature.

I wonder if we should then not even show this if the theme doesn't support it? It seems unexpected to me to have a feature that you might not be able to use? I wonder if there already is a check though?

Commit:
Iterate the text based on feedback

Commit:
Change text to be clearer

karmatosed commented:
I just did one more iteration based on last round of feedback. You can see the changes here:

![image](https://user-images.githubusercontent.com/253067/95989937-d0d18100-0e22-11eb-8995-9dbad935cc61.png)

kellychoffman commented:
:shipit:

youknowriad reviewed (approved):
LGTM code wise

youknowriad committed:
Fix test publish labels

karmatosed commented:
As the tests seem to have now passed, thanks to everyone for all the feedback and work in getting here. I am going to merge this! Excited to get this iteration in.

ZebulanStanphill commented:
> @ZebulanStanphill Thanks for your input. Regarding:
> 
> > We should probably note that "Use theme styles" may have no effect if the theme doesn't support the feature.
> 
> I wonder if we should then not even show this if the theme doesn't support it? It seems unexpected to me to have a feature that you might not be able to use? I wonder if there already is a check though?

Perhaps we could put `aria-disabled` on the control when it has no effect? Then again, it might be slightly annoying to not be able to enable/disable the feature until after you've switched to a theme that supports it. Perhaps we could just update the wording to something like "Use theme styles (if supported)".

mtias commented:
Nice to see this iteration! I think we need to look at better grouping next — for example, button labels in "General" doesn't make much sense. It could be better in "Appearance". Also the "Keyboard" grouping seems awkward to me. Also "theme styles" should be at the bottom in that group.

karmatosed commented:
@mtias no problem I can spin up an iterative PR to bring that in now this is in. I'll do that now.
